ql/indexes/inflationindex.cpp

```diff
@@ -70,23 +70,46 @@ namespace QuantLib {
                       Frequency frequency,
                       const Period& availabilityLag,
                       const Currency& currency,
-                      const Handle<ZeroInflationTermStructure>& zeroInflation)
-    : InflationIndex(familyName, region, revised, interpolated,
+                      const Handle<ZeroInflationTermStructure>& zeroInflation,
+                      bool forecastWhenPossible)
+        : InflationIndex(familyName, region, revised, interpolated,
                      frequency, availabilityLag, currency),
-      zeroInflation_(zeroInflation) {
+      zeroInflation_(zeroInflation),
+      forecastWhenPossible_(forecastWhenPossible) {
         registerWith(zeroInflation_);
     }
 
+    bool ZeroInflationIndex::forecastTodaysFixing(const Date& fixingDate) const {
+        return forecastWhenPossible_ && fixingDate >= zeroInflationTermStructure()->baseDate();
+    }
+
+
+    Rate ZeroInflationIndex::fixing(const Date& fixingDate) const {
+        return fixing(fixingDate, fixingDate);
+    }
+
     Rate ZeroInflationIndex::fixing(const Date& aFixingDate,
-                                    bool /*forecastTodaysFixing*/) const {
```

Comment:
If we ignore the changes that I want to make, you'll notice the `forecastTodaysFixing` variable is ignored. Is it OK if I remove this overload completely? Is there a case where a user will want to input this parameter? I can see only situations where it has to be calculated, as in this PR's case.